packages/common/src/api/tan-query/comments.ts

```diff
@@ -0,0 +1,719 @@
+
+import {
+  commentFromSDK,
+  replyCommentFromSDK,
+  transformAndCleanList
+} from '~/adapters'
+import { useAudiusQueryContext } from '~/audius-query'
+import { Comment, ID, ReplyComment } from '~/models'
+import { toast } from '~/store/ui/toast/slice'
+import { encodeHashId } from '~/utils'
+
+type CommentOrReply = Comment | ReplyComment
+
+const QUERY_KEYS = {
+  trackCommentList: 'trackCommentList',
+  comment: 'comment',
+  replies: 'replies'
+} as const
+
+const messages = {
+  loadError: (type: 'comments' | 'replies') =>
+    `There was an error loading ${type}. Please try again.`,
+  mutationError: (
+    actionType:
+      | 'pinning'
+      | 'unpinning'
+      | 'deleting'
+      | 'posting'
+      | 'editing'
+      | 'reacting to'
+      | 'reporting'
+  ) => `There was an error ${actionType} that comment. Please try again`,
+  muteUserError: 'There was an error muting that user. Please try again.'
+}
+
+/**
+ *
+ * QUERIES
+ *
+ */
+
+type GetCommentsByTrackArgs = {
+  trackId: ID
+  userId: ID | null
+  sortMethod: CommentSortMethod
+  pageSize?: number
+}
+
+export const useGetCommentsByTrackId = ({
+  trackId,
+  userId,
+  sortMethod,
+  pageSize = 5
+}: GetCommentsByTrackArgs) => {
+  const { audiusSdk, reportToSentry } = useAudiusQueryContext()
+  const isMutating = useIsMutating()
+  const queryClient = useQueryClient()
+  const queryRes = useInfiniteQuery({
+    enabled: !!userId && !!trackId && isMutating === 0,
+    getNextPageParam: (lastPage: ID[], pages) => {
+      if (lastPage?.length < pageSize) return undefined
+      return (pages.length ?? 0) * pageSize
+    },
+    queryKey: [QUERY_KEYS.trackCommentList, trackId, sortMethod],
+    queryFn: async ({ pageParam: currentPage = 0 }): Promise<ID[]> => {
+      const sdk = await audiusSdk()
+      const commentsRes = await sdk.tracks.trackComments({
+        trackId: encodeHashId(trackId),
+        offset: currentPage,
+        limit: pageSize,
+        sortMethod,
+        userId: userId?.toString() ?? undefined
+      })
+      const commentList = transformAndCleanList(
+        commentsRes.data,
+        commentFromSDK
+      )
+
+      // Populate individual comment cache
+      commentList.forEach((comment) => {
+        queryClient.setQueryData([QUERY_KEYS.comment, comment.id], comment)
+        comment?.replies?.forEach?.((reply: ReplyComment) =>
+          queryClient.setQueryData([QUERY_KEYS.comment, reply.id], reply)
+        )
+      })
+      // For the comment list cache, we only store the ids of the comments (organized by sort method)
+      return commentList.map((comment) => comment.id)
+    },
+    onError: (error: Error) => {
+      reportToSentry({
+        error,
+        name: 'Comments'
+      })
+      toast({ content: messages.loadError('replies') })
+    },
+    staleTime: Infinity, // Stale time is set to infinity so that we never reload data thats currently shown on screen (because sorting could have changed)
```

Comment:
I know you mentioned this in the front-end sync, but I'm not sure I follow exactly and want to understand better. Why does sorting changing require us to keep stale data?

Reply:
Short answer:
Tan-query refreshes stale data on window refocus and we want to keep comments from shifting on the user when they refocus due to the refreshed data potentially having a different sort order

Longer answer:
Context: for comments, product & design wanted to stay away from "live" refreshing comments because comments can rearrange based on the sort mechanisms (also note: sorting is computed on the backend)
The reason stale time is relevant here is because by default tan-query assumes all data is stale right after you fetch it. When your data is considered stale, tan-query will look for opportunities to refresh it for you; the main way we wanted to avoid here with comments is on window refocus.
Instead product wants to add a manual refresh button so the user can manually reload the page.

Reply:
I'm still not sure I fully grasp the implications of sort order on this (haven't played enough with comments probably) but would it not make sense to just turn `refetchOnWindowFocus` off?
https://tanstack.com/query/v4/docs/framework/react/guides/window-focus-refetching (got there from https://tanstack.com/query/v4/docs/framework/react/guides/important-defaults#:~:text=To%20change%20this%20behavior,%20you%20can%20configure%20your%20queries%20both)

Reply:
Yeah while I did single out the window focus one since that was the most noticeable to me (and the reason I knew to add the stale time in the first place) there are still more cases that we want to avoid - 
![image](https://github.com/user-attachments/assets/3751c869-51d9-4885-a3bf-05e6e8d28897)
1 and 4 there are moot points for comments how I have it now, but network reconnect is another one we want to avoid.

Setting the data to never stale just acts as a blanket approach to explicitly avoid any of those scenarios where the data refreshes on you.

This also matches our product spec since since we're taking the stance that the comment data should never refresh while active on the page without the user manually clicking the "reload comments" button that we're adding.

Also in general, my plan with these changes was to keep away from changing anything with the global `queryClient` and isolate stuff for comments as much as possible. We can re-evaluate our app-wide defaults later.

Hopefully that clears it up? But happy to explain it further.

Reply:
Thanks for the deetz! I'm very clearly ootl so this was helpful

> comment data should never refresh while active on the page without the user manually clicking the "reload comments" button that we're adding.

This is such an odd product decision to me